### PR TITLE
Make new directory under `federated` named `network`

### PR DESCRIPTION
**Pull request by Jakio815**
This PR is the first step abstracting the network layer which is in lf-lang/reactor-c#281.

This changes the file structure of the reactor-c, adding a new directory under `federated` named `networks`.

Currently only moved `net_util.c` `net_common.h` and `net_util.h` under newly named directory `networks`.

This is to add more files in future such as `net_tls.c`. 

The corresponding `CMakelists.txt` files have been modified together.

The idea comes from how the `platform` works. Each platform related header such as  `lf_arduino_support.h` `lf_rp2040_support.h` is inside the directory `platform`.

This must be merged together with https://github.com/lf-lang/lingua-franca/pull/2070

### CODE REVIEW

**Comment by Jakio815** [comment]
@lhstrh This was discussed in the security meeting. The point is just moving network related c and header files under a directory `network`. Do you think it's fine?

**Comment by lhstrh** [comment]
I would call `network` or `net`, not `networks`, but the idea seems fine. Let me request review from @erlingrj.

**Comment by erlingrj** [comment]
Looks like a reasonable approach. Just one question about the new CMakeLists.txt

File: core/federated/networks/CMakeLists.txt [hunk 1]
```diff
@@ -0,0 +1,5 @@
+set(LF_NETWORK_FILES net_util.c)
```
[the remark this answers is not in the document]
Comment (Jakio815) [reply core/federated/networks/CMakeLists.txt:1]: @erlingrj I'm not really an expert of CMake, so not really sure what's your question. I followed `reactor-c/core/platform/CMakeLists.txt`. Is there something wrong?

**Comment by Jakio815** [comment]
> I would call `network` or `net`, not `networks`, but the idea seems fine. Let me request review from @erlingrj.

Fixed all to `network`

File: core/CMakeLists.txt [hunk 2]
```diff
@@ -18,6 +18,7 @@ add_library(core ${GENERAL_SOURCES})
 # Add sources for either threaded or unthreaded runtime
 if (DEFINED FEDERATED)
     include(federated/CMakeLists.txt)
+    include(federated/network/CMakeLists.txt)
```
Comment (Jakio815) [inline core/CMakeLists.txt:21]: @erlingrj I think I kind of understood your question. I added it here!

**Comment by Jakio815** [comment]
@lhstrh Can I merge this together with https://github.com/lf-lang/lingua-franca/pull/2070?

**Comment by lhstrh** [comment]
I noticed that not all test in https://github.com/lf-lang/lingua-franca/pull/2070 are passing though...

File: lingua-franca-ref.txt [hunk 3]
```diff
@@ -1 +1 @@
-master
+47b87a49ceece1f0065a9dbf9b8a06c9fcdfe01d
```
Comment (lhstrh) [inline lingua-franca-ref.txt:1]:
```suggestion
network-structure
```

Comment (lhstrh) [reply lingua-franca-ref.txt:1]: Please make sure to test against the latest in commit in the `network-structure` branch.

**Review by lhstrh — COMMENTED** [review]
Looks good modulo comment.

**Review by lhstrh — APPROVED** [review]

**Comment by edwardalee** [comment]
As long as you are using reactor-c/main it is OK to merge this.

**Pull request merged**